pandas/io/parsers.py

```diff
@@ -2251,6 +2314,113 @@ def _maybe_parse_dates(self, values, index, try_parse_dates=True):
         return values
 
 
+class BytesIOWrapper:
+    """
+    Allows the pyarrow engine for read_csv() to read from string buffers
+    """
+
+    def __init__(self, string_buffer: StringIO, encoding: str = "utf-8"):
+        self.string_buffer = string_buffer
+        self.encoding = encoding
+
+    def __getattr__(self, attr: str):
+        return getattr(self.string_buffer, attr)
+
+    def read(self, size: int = -1):
+        content = self.string_buffer.read(size)
+        return content.encode(self.encoding)
+
+
+class ArrowParserWrapper(ParserBase):
```

Comment:
you will need to refactor this as the current code is very different from this.

Also I really don't like doing all of this validation in a single function.

Reply:
> you will need to refactor this as the current code is very different from this.

Can you clarify a bit more what you mean? Or point to recent changes related to this?